appstudio-utils/util-scripts/cosign-verify.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Runs cosign verify to verify the signature of the provided image
+# reference with the provided public key
+# usage:
+#   cosign-verify.sh <image reference> <public key> <output file>
+# where:
+#   <image reference> string pointing to the image, e.g.
+#                     registry-host/image:tag
+#   <public key>      PEM-encoded string or a valid cosign key reference
+#                     e.g. '-----BEGIN PUBLIC KEY-----\n...' or
+#                     k8s://default/testsecret
+#   <output file>     where to store the result in JSON format
+
+cosign-verify() {
+  local image_reference="$1"
+
+  local public_key="$2"
+  if [[ "${public_key}" == -----BEGIN* ]]; then
+    local tmp_public_key
+    tmp_public_key="$(mktemp)"
+    trap 'rm '"${tmp_public_key}" EXIT
+    echo "${public_key}" > "${tmp_public_key}"
+    public_key="${tmp_public_key}"
+  fi
+
+  local output_file="$3"
+
+  cosign verify \
+    --key "${public_key}" \
+    --output-file "${output_file}" \
+    "${image_reference}"
+}
+
+# When included from shellspec we don't want to invoke cosign-verify
+${__SOURCED__:+return}
```

Comment:
I'm confused here, because of the `echo "cosign $*"` don't we expect shellspec does run cosign, or at least its redefined version of it?